### PR TITLE
[dagster-airlift] AirflowInstance

### PR DESCRIPTION
**Pull request by dpeng817**
Add AirflowInstance API, which vends out access to the airflow rest API.

This paves the way for having multiple auth backends supporting airlift utilities. Also just happens to feel nicer api-wise.

### CODE REVIEW

**Comment by dpeng817** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23343?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#23344** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23344?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23343** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23343?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#23326** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23326?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23324** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23324?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23300** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23300?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @dpeng817 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py [hunk 1]
```diff
@@ -43,98 +44,172 @@ class TaskMapping(NamedTuple):
...
+        self.username = username
+        self.password = password
+
+    def get_session(self) -> requests.Session:
+        session = requests.Session()
+        session.auth = (self.username, self.password)
+        return session
+
+
+class AirflowInstance(NamedTuple):
```
Comment (schrockn) [inline examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:63]: seems a little goofy to have this be a NamedTuple but nbd

**Review by schrockn — APPROVED** [review]
Makes sense. This will be a nice seem for testability as we can make a fake for it.

File: examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py (hunk 1 above)
Comment (dpeng817) [reply examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:63]: what's goofy about it being a namedtuple? I guess just vs a regular class you mean?

Honestly it's just learned behavior from the codebase at this point

Comment (schrockn) [reply examples/experimental/dagster-airlift/dagster_airlift/airflow_utils.py:63]: Generally NamedTuple are "dumb" value objects that represent a value/record at a point in time. This is a more complex object that will likely accumulate state, require caching, that sort of thing.

**Pull request merged**